tests/conftest.py

```diff
@@ -12,3 +21,240 @@ def api_test_config() -> APITestSessionConfig:
+
+
+@pytest.fixture(scope="session")
+def token_url():
+    oauth_proxy = get_env("OAUTH_PROXY")
+    oauth_base_uri = get_env("OAUTH_BASE_URI")
+    return f"{oauth_base_uri}/{oauth_proxy}/token"
+
+
+@pytest.fixture
+def make_product(environment, service_name):
+    async def _make_product(product_scopes):
+        # Setup
+        product = ApigeeApiProducts()
+        await product.create_new_product()
+
+        print(f"CREATED PRODUCT NAME: {product.name}")
+
+        # Update products allowed paths
+        proxies = [f"identity-service-mock-{environment}"]
+
+        if service_name is not None:
+            proxies.append(service_name)
+
+        await product.update_proxies(proxies)
+        await product.update_scopes(scopes=product_scopes)
+        return product
+
+    return _make_product
+
+
+@pytest.fixture
+def make_app():
+    async def _make_app(product, custom_attributes):
+        # Setup
+        app = ApigeeApiDeveloperApps()
+        await app.create_new_app()
+
+        print(f"CREATED APP NAME: {app.name}")
+
+        await app.set_custom_attributes(custom_attributes)
+
+        # Assign the new app to the product
+        await app.add_api_product([product.name])
+
+        app.oauth = OauthHelper(app.client_id, app.client_secret, app.callback_url)
+        return app
+
+    return _make_app
+
+
+@pytest.fixture
+async def patient_care_product(make_product):
+    # Setup
+    product = await make_product(["urn:nhsd:apim:user-nhs-login:P9:e-referrals-service-patient-care-api"])
+
+    await product.update_attributes(
+        attributes={"requestLimit_referral": "100",
+            "requestLimit_general": "100",
+            "burstLimit_referral": "1200pm",
+            "burstLimit_general": "1200pm"
+        }
+    )
+
+    yield product
+
+    # Teardown
+    print(f"Cleanup product: {product.name}")
+    await product.destroy_product()
+
+
+@pytest.fixture
+async def patient_care_app(
+    make_app,
+    patient_care_product
+):
+    # Setup
+    app = await make_app(
+        patient_care_product,
+        {
+            "jwks-resource-url": __JWKS_RESOURCE_URL
+        },
+    )
+
+    yield app
+
+    # Teardown
+    print(f"Cleanup app: {app.name}")
+    await app.destroy_app()
+
+
+@pytest.fixture
+async def nhs_login_subject_token(patient_care_app: ApigeeApiDeveloperApps):
+    id_token_claims = {
+        "aud": "tf_-APIM-1",
+        "id_status": "verified",
+        "token_use": "id",
+        "auth_time": 1616600683,
+        "iss": "https://internal-dev.api.service.nhs.uk",  # Points to internal dev -> testing JWKS
+        "sub": "https://internal-dev.api.service.nhs.uk",
+        "exp": int(time()) + 300,
+        "iat": int(time()) - 10,
+        "vtm": "https://auth.sandpit.signin.nhs.uk/trustmark/auth.sandpit.signin.nhs.uk",
+        "jti": str(uuid4()),
+        "identity_proofing_level": "P9",
+        "birthdate": "1939-09-26",
+        "nhs_number": "9912003888",
```

Comment:
I wonder if it would be worth adding the nhs number provided as a parameter to this fixture? As all tests are running currently with the same patient however its probably not necessary currently.